Added support for cluster roll.

sharadkesarwani opened the pull request:
Added support for cluster roll.

# Jira Ticket
 https://spotinst.atlassian.net/browse/SPOTAUT-17448

sharadkesarwani committed:
Added support for cluster roll.

Added support for cluster roll.

sharadkesarwani committed:
Added documentation and support for node_pool_names

Added documentation and support for node_pool_names

chandra1-n reviewed (changes requested):
please add roll in tests as well and separate roll for vng as well

sharadkesarwani committed:
Corrected package name

Corrected package name

sharadkesarwani committed:
Added support to trigger roll on VNG updation

Added support to trigger roll on VNG updation

sharadkesarwani committed:
Removed conditioned_roll_params support and filters from predefined list for both cluster and vng roll

Removed conditioned_roll_params support and filters from predefined list for both cluster and vng roll

sharadkesarwani committed:
Merge branch 'main' into SPOTAUT-17448-aks-cluster-roll

sharadkesarwani committed:
Executed dependency check commands

Executed dependency check commands

chandra1-n commented on CHANGELOG.md:
```diff
@@ -1,4 +1,7 @@
 ## Unreleased
+## 1.159.0 (January, 15 2024)
```
Update the version and date

sharadkesarwani replied:
updated version to latest 1.162.0, and current date February, 15 2024

chandra1-n commented on CHANGELOG.md:
```diff
@@ -1,4 +1,7 @@
 ## Unreleased
+## 1.159.0 (January, 15 2024)
+ENHANCEMENTS:
+* resource/spotinst_ocean_aks_np: Added support to trigger cluster roll on attribute updation and introduced `conditioned_roll_params` field to customized attribute modification to trigger cluster roll.
```
remove conditioned_roll_params, we're not giving that option for now

sharadkesarwani replied:
Removed

chandra1-n commented on docs/resources/ocean_aks_np.md:
```diff
@@ -276,3 +276,35 @@ The following arguments are supported:
+## Update Policy
+
+* `update_policy` - (Optional)
+  * `should_roll` - (Required) Enables the roll.
```
add - if set to true along with the cluster update, roll will be triggered

sharadkesarwani replied:
Added

chandra1-n commented on docs/resources/ocean_aks_np.md:
```diff
@@ -276,3 +276,35 @@ The following arguments are supported:
+
+* `update_policy` - (Optional)
+  * `should_roll` - (Required) Enables the roll.
+  * `conditioned_roll` - (Optional, Default: false) Spot will perform a cluster Roll in accordance with a relevant modification of the cluster’s settings. When set to true , only specific changes in the cluster’s configuration will trigger a cluster roll (such as AMI, Key Pair, user data, instance types, load balancers, etc).
```
such as examples are same as ocean aws, please modify according to ocean aks

sharadkesarwani replied:
Updated as per AKS

chandra1-n commented on docs/resources/ocean_aks_np.md:
```diff
@@ -276,3 +276,35 @@ The following arguments are supported:
+* `update_policy` - (Optional)
+  * `should_roll` - (Required) Enables the roll.
+  * `conditioned_roll` - (Optional, Default: false) Spot will perform a cluster Roll in accordance with a relevant modification of the cluster’s settings. When set to true , only specific changes in the cluster’s configuration will trigger a cluster roll (such as AMI, Key Pair, user data, instance types, load balancers, etc).
+  * `conditioned_roll_params` - (Optional) A custom list of attributes will trigger the cluster roll operation (overrides the predefined list of parameters). Valid only when the `conditioned_roll` parameter is set to true. (Valid values: `"availability_zones"`, `"max_pods_per_node"`, `"enable_node_public_ip"`, `"os_disk_size_gb"`, `"os_disk_type"`, `"os_sku"`, `"kubernetes_version"`, `"vnet_subne ...
```
please remove as we're not giving this option now

sharadkesarwani replied:
Removed

chandra1-n commented on docs/resources/ocean_aks_np.md:
```diff
@@ -276,3 +276,35 @@ The following arguments are supported:
+  * `should_roll` - (Required) Enables the roll.
+  * `conditioned_roll` - (Optional, Default: false) Spot will perform a cluster Roll in accordance with a relevant modification of the cluster’s settings. When set to true , only specific changes in the cluster’s configuration will trigger a cluster roll (such as AMI, Key Pair, user data, instance types, load balancers, etc).
+  * `conditioned_roll_params` - (Optional) A custom list of attributes will trigger the cluster roll operation (overrides the predefined list of parameters). Valid only when the `conditioned_roll` parameter is set to true. (Valid values: `"availability_zones"`, `"max_pods_per_node"`, `"enable_node_public_ip"`, `"os_disk_size_gb"`, `"os_disk_type"`, `"os_sku"`, `"kubernetes_version"`, `"vnet_subne ...
+  * `roll_config` - (Required) While used, you can control whether the group should perform a deployment after an update to the configuration.
```
re-arrange the attributes under roll_config

sharadkesarwani replied:
Rearranged

chandra1-n commented on docs/resources/ocean_aks_np.md:
```diff
@@ -276,3 +276,35 @@ The following arguments are supported:
+update_policy {
+  should_roll = false
+  conditioned_roll = true
+  conditioned_roll_params = ["os_disk_type", "availability_zones"]
```
removed conditioned_roll_params

sharadkesarwani replied:
Removed

chandra1-n commented on docs/resources/ocean_aks_np.md:
```diff
@@ -276,3 +276,35 @@ The following arguments are supported:
+
+  roll_config {
+    batch_size_percentage = 33
+    vng_ids = ["ols-1a2b3c4d"]
```
remove vng_ids

sharadkesarwani replied:
removed from cluster roll example.

chandra1-n reviewed (changes requested):
update the document for aks-np-vng as well

patrapritish1 committed:
commented vnet subnet ids from aks 2.0 test

patrapritish1 committed:
resolved conflicts with main

sharadkesarwani committed:
Updated documentation, change log and fixed setting value of VngId, NodePoolNames and NodeNames.

Updated documentation, change log and fixed setting value of VngId, NodePoolNames and NodeNames.

sharadkesarwani committed:
Merge branch 'SPOTAUT-17448-aks-cluster-roll' of https://github.com/spotinst/terraform-provider-spotinst into SPOTAUT-17448-aks-cluster-roll

sharadkesarwani committed:
Updated dependency

Updated dependency

chandra1-n commented on docs/resources/ocean_aks_np.md:
```diff
@@ -276,3 +276,32 @@ The following arguments are supported:
+  * `should_roll` - (Required) If set to true along with the cluster update, roll will be triggered.
+  * `conditioned_roll` - (Optional, Default: false) Spot will perform a cluster Roll in accordance with a relevant modification of the cluster’s settings. When set to true , only specific changes in the cluster’s configuration will trigger a cluster roll (such as availability_zones, max_pods_per_node, enable_node_public_ip, os_disk_size_gb, os_disk_type, os_sku, kubernetes_version, vnet_subnet_id ...
+  * `roll_config` - (Required) While used, you can control whether the group should perform a deployment after an update to the configuration.
+    * `batch_min_healthy_percentage` - (Optional, Default: 50) Indicates the threshold of minimum healthy instances in single batch. If the amount of healthy instances in single batch is under the threshold, the cluster roll will fail. If exists, the parameter value will be in range of 1-100. In case of null as value, the default value in the backend will be 50%. Value of param should represent t ...
```
please re-validate the description with aks attributes

sharadkesarwani replied:
Updated description as per AKS

chandra1-n commented on docs/resources/ocean_aks_virtual_node_group.md:
```diff
@@ -99,3 +99,31 @@ The following arguments are supported:
+* `update_policy` - (Optional)
+  * `should_roll` - (Required) If set to true along with the vng update, roll will be triggered.
+  * `conditioned_roll` - (Optional, Default: false) Spot will perform a cluster Roll in accordance with a relevant modification of the cluster’s settings. When set to true , only specific changes in the cluster’s configuration will trigger a cluster roll (such as availability_zones, max_pods_per_node, enable_node_public_ip, os_disk_size_gb, os_disk_type, os_sku, kubernetes_version, vnet_subnet_id ...
+  * `roll_config` - (Required) While used, you can control whether the group should perform a deployment after an update to the configuration.
```
in the code it is defined as Optional, please re-validate

sharadkesarwani replied:
Updated to optional.

chandra1-n commented on docs/resources/ocean_aks_np.md:
```diff
@@ -276,3 +276,32 @@ The following arguments are supported:
+* `update_policy` - (Optional)
+  * `should_roll` - (Required) If set to true along with the cluster update, roll will be triggered.
+  * `conditioned_roll` - (Optional, Default: false) Spot will perform a cluster Roll in accordance with a relevant modification of the cluster’s settings. When set to true , only specific changes in the cluster’s configuration will trigger a cluster roll (such as availability_zones, max_pods_per_node, enable_node_public_ip, os_disk_size_gb, os_disk_type, os_sku, kubernetes_version, vnet_subnet_id ...
+  * `roll_config` - (Required) While used, you can control whether the group should perform a deployment after an update to the configuration.
```
in the code it is defined as Optional, please re-validate

sharadkesarwani replied:
Updated to optional.

chandra1-n reviewed (changes requested):
please fix the review comments

sharadkesarwani committed:
Modified code to make `BatchSizePercentage` as optional and updated document.

Modified code to make `BatchSizePercentage` as optional and updated document.
